lib/beacon_api/controllers/v2/beacon_controller.ex

```diff
@@ -61,14 +63,31 @@ defmodule BeaconApi.V2.BeaconController do
       version: "deneb",
       execution_optimistic: true,
       finalized: false,
-      data: %{
-        # TODO: return block as JSON
-        message: inspect(block)
-      }
+      data: to_json(block)
     })
   end
 
-  defp block_not_found(conn) do
+  @spec to_json(any()) :: any()
+  def to_json(map) when is_map(map) do
+    map
+    |> Map.from_struct()
+    |> Stream.map(&to_json/1)
+    |> Map.new()
+  end
+
+  def to_json(list) when is_list(list), do: Enum.map(list, &to_json/1)
+
+  def to_json({:aggregation_bits, v}) do
```

Comment:
we probably want a generic way of encoding a bitlist instead of having to specify the field (in this case `aggregation_bits`).

Reply:
Maybe we want to do something similar to the `encode` function https://github.com/lambdaclass/lambda_ethereum_consensus/blob/main/lib/types/beacon_chain/beacon_state.ex#L152

Reply:
Generic/recursive would be ideal!